apps/meteor/tests/e2e/e2e-encryption.spec.ts

```diff
@@ -654,6 +654,68 @@ test.describe.serial('e2ee room setup', () => {
 		await expect(poHomeChannel.content.lastUserMessage.locator('.rcx-icon--name-key')).toBeVisible();
 	});
 
+	test('expect save password state on encrypted DM', async ({ page }) => {
+		await page.goto('/account/security');
+		await poAccountProfile.securityE2EEncryptionSection.click();
+		await poAccountProfile.securityE2EEncryptionResetKeyButton.click();
+
+		await page.locator('role=button[name="Login"]').waitFor();
+
+		await page.reload();
+
+		await page.locator('role=button[name="Login"]').waitFor();
+
+		await injectInitialData();
+		await restoreState(page, Users.admin);
+
+		await page.goto('/home');
+
+		await page.locator('role=banner >> text="Save your encryption password"').waitFor();
+		await expect(page.locator('role=banner >> text="Save your encryption password"')).toBeVisible();
+
+		await poHomeChannel.sidenav.openNewByLabel('Direct message');
+		await poHomeChannel.sidenav.inputDirectUsername.click();
+		await page.keyboard.type('user2');
+		await page.waitForTimeout(1000);
+		await page.keyboard.press('Enter');
+		await poHomeChannel.sidenav.btnCreate.click();
+
+		await expect(page).toHaveURL(`/direct/rocketchat.internal.admin.testuser2`);
+
+		await poHomeChannel.tabs.kebab.click({ force: true });
+		await expect(poHomeChannel.tabs.btnEnableE2E).toBeVisible();
+		await poHomeChannel.tabs.btnEnableE2E.click({ force: true });
+
+		await poHomeChannel.content.encryptedRoomHeaderIcon.first().waitFor();
+		await expect(poHomeChannel.content.encryptedRoomHeaderIcon.first()).toBeVisible();
+
+		await page.locator('role=button[name="Save E2EE password"]').waitFor();
+		await expect(page.locator('role=button[name="Save E2EE password"]')).toBeVisible();
+
+		await poHomeChannel.tabs.btnE2EERoomSetupDisableE2E.waitFor();
+		await expect(poHomeChannel.tabs.btnE2EERoomSetupDisableE2E).toBeVisible();
+		await expect(poHomeChannel.tabs.btnTabMembers).not.toBeVisible();
+		await expect(poHomeChannel.tabs.btnRoomInfo).not.toBeVisible();
+
+		await expect(poHomeChannel.content.inputMessage).not.toBeVisible();
+
+		await page.locator('role=button[name="Save E2EE password"]').click();
```

Comment:
This could be the same hypothetical variable I mentioned on comment on line 692. You already had 2 locators for it, maybe it's better to save it in a const so you only need to find the element once

Reply:
Makes sense.